extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQWorkerTask.java

```diff
@@ -185,4 +189,15 @@ public int hashCode()
   {
     return Objects.hash(super.hashCode(), controllerTaskId, workerNumber, retryCount, worker);
   }
+
+  @Override
+  public LookupLoadingSpec getLookupLoadingSpec()
+  {
+    if (getContext().containsKey(PlannerContext.CTX_LOOKUPS_TO_LOAD)) {
```

Comment:
> I think this method of getting something from context and getting a lookupLoading spec should be in an utility class since this would be used in multiple places.

I think this is a non-problem in the newer approach, since MSQControllerTask has been updated to return NONE, and MSQWorkerTask also has some modifications in the logic. We can refactor it into a common place if needed in future though.

> I donot see the context key's being passed to the worker.

I could see lookups being loaded selectively on the worker nodes. I also verified the values in the worker payload as we discussed earlier today. With that said, the PR has been updated to the new approach we discussed, so please have a look at the latest code changes once. Thanks!